yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -899,8 +938,21 @@ object Client extends Logging {
   // Distribution-defined classpath to add to processes
   val ENV_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH"
 
-  // Subdirectory where the user's hadoop config files will be placed.
-  val LOCALIZED_HADOOP_CONF_DIR = "__hadoop_conf__"
+  // Subdirectory where the user's Spark and Hadoop config files will be placed.
+  val LOCALIZED_CONF_DIR = "__spark_conf__"
+
+  // Name fo the file in the conf archive containing Spark configuration.
+  val SPARK_CONF_FILE = "__spark_conf__.properties"
+
+  // Subdirectory where the user's python files (not archives) will be placed.
+  val LOCALIZED_PYTHON_DIR = "__pyfiles__"
+
+  // Key in SparkConf where to find the executors' PYTHONPATH. This cannot be set using
+  // `SparkConf.setExecutorEnv`, because that would cause it to propagate to the python
+  // code via the configuration, and then override the process's own environment when
+  // launching workers. Since it contains variables that are expanded by YARN, that cannot
+  // happen.
```

Comment:
Actually, re-tested without this and it seems like using the property file handles the original issue already. So I'll just remove it.